pkg/flavors/postgres/postgres.go

```diff
@@ -0,0 +1,264 @@
+// Copyright 2020 Tetrate
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package postgres
+
+import (
+	"bytes"
+	"fmt"
+	"net"
+	"strings"
+	"text/template"
+
+	valid "github.com/asaskevich/govalidator"
+	"github.com/tetratelabs/getenvoy/pkg/flavors"
+)
+
+// Define template parameter names
+const endpoint string = "endpoint"
+const inport string = "inport"
+
+// Flavor implements flavor.FlavorConfigTemplate interface
+// and stores config data specific to Postgres template.
+type Flavor struct {
+	// Location of the postgres server
+	endpoints []*clusterEndpoint
+	// Envoy's listener port
+	InPort      string
+	ClusterType string
+}
+
+type clusterEndpoint struct {
+	EndpointAddr string
+	EndpointPort string
+	IsIP         bool
+}
+
+var flavor = Flavor{
+	InPort: "5432",
+}
+
+func init() {
+	// Register postgres flavor.
+	flavors.AddFlavor("postgres", &flavor)
+}
+
+// GenerateConfig method takes command line parameters and creates
+// Postgres specific Envoy config.
+func (f *Flavor) GenerateConfig(params map[string]string) (string, error) {
+	var err error
+	var config string
+
+	err = f.parseInputParams(params)
+	if err != nil {
+		return "", err
+	}
+	// Now process the main template. This includes everything except endpoints.
+	// NOw run the template substitution
+	mainConfig, err := f.generateMainConfig()
+	if err != nil {
+		return "", err
+	}
+	config += mainConfig
+
+	endpointsConfig, err := f.generateEndpointSetConfig()
+	if err != nil {
+		return "", err
+	}
+	config += endpointsConfig
+
+	return config, nil
+}
+
+// Utility functions.
+
+// Function parses and verifies inout params for consistency
+// and correctness.
+func (f *Flavor) parseInputParams(params map[string]string) error {
+	var err error
+
+	for param, value := range params {
+		switch param {
+		case endpoint:
+			f.endpoints, err = parseEndpointSet(value)
+			if err != nil {
+				return err
+			}
+		case inport:
+			if !valid.IsInt(value) {
+				return fmt.Errorf("Value for templateArg %s must be integer number", param)
+			}
+			f.InPort = value
+		default:
+			fmt.Printf("Ignoring unrecognized template parameter: %s", param)
+		}
+	}
+
+	// Check if all required params have been found in the parameter list
+	if len(f.endpoints) == 0 {
+		return fmt.Errorf("endpoint parameter must be specified")
```

Comment:
I rephrased it. It started with lowercase because it refers to command line parameter "endpoint". I changed it to "Parameter endpoints must be specified:.